resources/views/partials/item.blade.php

```diff
@@ -68,14 +68,14 @@
             <a href="https://{{ $wowheadLocale . $wowheadSubdomain }}.wowhead.com/item={{ $itemId }}" target="_blank" class="{{ isset($itemQuality) ? $itemQuality : '' }}">{{ $itemName }}</a>
         @elseif (isset($guild) && $guild)
             @if (isset($auditLink) && $auditLink)
-                <a href="{{ route('guild.auditLog', ['guildId' => $guild->id, 'guildSlug' => $guild->slug, 'item_id' => $itemId]) }}"
+                <a href="{{ route('guild.auditLog', ['guildId' => $guild->id, 'guildSlug' => $guild->slug, 'itemId' => $itemId]) }}"

```

Comment:
`item_id` is a GET parameter. We want to keep it as `item_id`.

Reply:
Roger, did not know that will check it out!

Reply:
Not sure if there's a good way to show that distinction in these `route()` calls. The GET parameters just blend in with the ones that are there because the route needs them.

Reply:
Where this specific parameter gets used: https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/AuditLogController.php#L173

Reply:
So if I understand correctly this is the only place where you actually use item_id as a get parameter (the `AuditLogController`)?

As for distinction between route parameters and get parameters you _could_ omit the names of route parameters e.g. `{{ route('guild.auditLog', [$guild->id, $guild->slug, 'itemId' => $itemId]) }}`

Am I really sold on that idea? Not really.. I'd much rather just remove the guild parameters entirely and let the instance inject it based on whatever guild is decided upon in the middleware. That should work perfectly fine unless you can audit/access someone else's guild details of course where `guildId` points towards an id that differs from your own guild.

What caused an extra level of confusion for me is that `item_id` and `item.id` are different things. I also  noticed that `item_id` is not unique and there are many instances of duplicate `item_id` values in the database (at least in mine). If for whatever reason you'd ever have two identical `item_id`s pointing to two different `id`s then things would officially get confusing beyond the point of no return. Out of curiosity: is there any valid use case for ever not having non-duplicate `item_id`s?

Reply:
Sorry, I missed this one particular comment for whatever reason. Just happened to see it while scrolling through some old email notifications before archiving them.

> So if I understand correctly this is the only place where you actually use item_id as a get parameter (the AuditLogController)?

Yes, I think so.

> I'd much rather just remove the guild parameters entirely and let the instance inject it based on whatever guild is decided upon in the middleware.

Yeah I'd rather not have the two parameters copied everywhere too. It's just what I did to get it done. They weren't originally there, but I added them in after the fact... at some point while migrating away from being an app solely for my guild and into being an app for the general audience and allowing people to belong to multiple guilds.

> I also noticed that item_id is not unique and there are many instances of duplicate item_id values in the database (at least in mine).

`item_id` was what the field was originally named in the SQL database from the WoW client. For whatever reason, I kept it as such. It wasn't a problem at first.

But when I added TBC support, and the TBC database had all of the Classic items in it as well, and I didn't want to merge the two databases (I wanted the TBC version of a Classic item to remain separate from the original version of that item, we now get duplicates. So I added in `id` as the primary key, and use `expansion_id` combined with `item_id` in lookups to create uniqueness)... though I don't believe it's a composite key.

Given that `id` didn't exist at first, a lot of the foreign keys are based off of `item_id`, but it still seems to work out ok... maybe I added `WHERE item_id = ? AND expansion_id = 1` to all of the queries to eliminate the problem in some cases... but not sure how the FK determines which copy to use. Just a toss-up I guess.

> Out of curiosity: is there any valid use case for ever not having non-duplicate item_ids?
I don't follow. `id` is always unique... `item_id` (ingame ID and the one we use for wowhead) will be duplicated when the same item appears in multiple expansions.

If I reword your question the other way around "is there any valid use case for ever having duplicate `item_id`s", I still don't follow. Maybe the previous information answered your question? If not, please reword it.

Reply:
Yeah it makes more sense now. What threw me off was that (at least in my database) the same item_id is used 4 times or sometimes even more. A while ago I started working on a crawler that combines the data from wowhead and the official Blizzard api for all languages. I need it for a project of my own but if at some point you want to clean up / normalize the item database a bit more (or add Italian translations, which are not present on wowhead) then be sure to hook me up.